ApplicationInsights/JavaScriptSDK/Telemetry/PageViewManager.ts

```diff
@@ -44,74 +47,82 @@ export class PageViewManager {
     *
     * In all cases page view performance is sent once (only for the 1st call of trackPageView), or not sent if navigation timing is not supported.
     */
-    public trackPageView(name?: string, url?: string, properties?: Object, measurements?: Object, duration?: number) {
-        // ensure we have valid values for the required fields
-        if (typeof name !== "string") {
-            name = window.document && window.document.title || "";
+    public trackPageView(pageView: IPageViewTelemetry, customProperties?: { [key: string]: any }) {
+        let name = pageView.name;
+        if (CoreUtils.isNullOrUndefined(name) || typeof name !== "string") {
+            pageView.name = window.document && window.document.title || "";
         }
 
-        if (typeof url !== "string") {
-            url = window.location && window.location.href || "";
+        let uri = pageView.uri;
+        if (CoreUtils.isNullOrUndefined(uri) || typeof uri !== "string") {
+            pageView.uri = window.location && window.location.href || "";
         }
 
-        var pageViewSent = false;
-        var customDuration = undefined;
-
-        if (PageViewPerformance.isPerformanceTimingSupported()) {
-            var start = PageViewPerformance.getPerformanceTiming().navigationStart;
-            customDuration = PageViewPerformance.getDuration(start, +new Date);
-
-            if (!PageViewPerformance.shouldCollectDuration(customDuration)) {
-                customDuration = undefined;
-            }
-        } else {
+        // if performance timing is not supported by the browser, send the page view telemetry with the duration provided by the user. If the user
+        // do not provide the duration, set duration to undefined
+        if (!PageViewPerformance.isPerformanceTimingSupported()) {
             this.appInsights.sendPageViewInternal(
-                name,
-                url,
-                !isNaN(duration) ? duration : undefined,
-                properties,
-                measurements);
+                pageView,
+                customProperties);
             this._channel.flush();
-            pageViewSent = true;
```

Comment:
You need this here, since pageView has been sent.

Reply:
discussed with Piyali offline.

---
In reply to: [205509477](https://github.com/Microsoft/ApplicationInsights-JS/pull/635#discussion_r205509477) [](ancestors = 205509477)